### PR TITLE
Created Knuth_Morris_Pratt.py

### PR DESCRIPTION
**Pull request by Yash-Thakare22**
Implemented of Knuth Morris Pratt Algorithm in Python

### CODE REVIEW

File: Python/cp/Knuth_Morris_Pratt.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+def KMP_Search(pattern, text):
+    a = len(text)
+    b = len(pattern)
+    prefix = prefix_arr(pattern, b)
+  
+    initial = []
+    m = 0
+    n = 0
+  
+    while m != a:
```
Comment (SKAUL05) [inline Python/cp/Knuth_Morris_Pratt.py:11]: Use meaningful and easy to understand variable names.

File: Python/cp/Knuth_Morris_Pratt.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            n = prefix[n-1]
+       
+        if n == b:
+            initial.append(m-n)
+            n = prefiX[n-1]
+        elif n == 0:
+            m += 1
+   
+    return initial
+def prefix_arr(pattern, b):
```
Comment (SKAUL05) [inline Python/cp/Knuth_Morris_Pratt.py:27]: Include appropriate Documentation (docstrings) for functions.

File: Python/cp/Knuth_Morris_Pratt.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Implementation of Knuth Morris Pratt Algorithm in Python
+def KMP_Search(pattern, text):
```
Comment (SKAUL05) [inline Python/cp/Knuth_Morris_Pratt.py:2]: Add Line break before function definition.

**Review by SKAUL05 — COMMENTED** [review]
You can check following links for documentation and style guiding
- [Sample Code NeoAlgo](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/math/Area.py)
- [PEP8 Guidelines](https://www.python.org/dev/peps/pep-0008/)

**Comment by atarax665** [comment]
@YashThakare Revert ASAP and do the suggested changes.

File: Python/cp/Knuth_Morris_Pratt.py (hunk 1 above)
Comment (Yash-Thakare22) [reply Python/cp/Knuth_Morris_Pratt.py:11]: For this algorithm this variables are meaningful variables

File: Python/cp/Knuth_Morris_Pratt.py (hunk 2 above)
Comment (Yash-Thakare22) [reply Python/cp/Knuth_Morris_Pratt.py:27]: Its proper as per the algorithm of KMP

File: Python/cp/Knuth_Morris_Pratt.py (hunk 3 above)
Comment (Yash-Thakare22) [reply Python/cp/Knuth_Morris_Pratt.py:2]: Okay I will

**Pull request closed without merging**